data/lib/api/match/match_model.dart

```diff
@@ -137,6 +138,8 @@ class MatchPlayer with _$MatchPlayer {
 
 @freezed
 class AddEditMatchRequest with _$AddEditMatchRequest {
+  // ignore: invalid_annotation_target
+  @JsonSerializable(anyMap: true, explicitToJson: true)
```

Comment:
Why do we need to use this?

Reply:
Match model contains another model as a field which toJson method is not converting all of them to json and that gives error when we update data to firebase as it accepts json only, to convert all underlying models to Json, we have used jsonSerializable.